feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -316,21 +312,94 @@ fun PreviewDisplay(
+                    implementationMode = implementationMode,
+                    onRequestWindowColorMode = onRequestWindowColorMode
+                )
+
+                val coordinateTransformer = remember { MutableCoordinateTransformer() }
                 CameraXViewfinder(
-                    modifier = Modifier.fillMaxSize(),
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .pointerInput(Unit) {
+                            detectTapGestures(
+                                onDoubleTap = { offset ->
+                                    // double tap to flip camera
+                                    Log.d(TAG, "onDoubleTap $offset")
+                                    onFlipCamera()
+                                },
+                                onTap = {
+                                    with(coordinateTransformer) {
+                                        val surfaceCoords = it.transform()
+                                        Log.d(
+                                            "TAG",
+                                            "onTapToFocus: " +
+                                                "input{$it} -> surface{$surfaceCoords}"
+                                        )
+                                        onTapToFocus(surfaceCoords.x, surfaceCoords.y)
+                                    }
+                                }
+                            )
+                        },
                     surfaceRequest = it,
-                    implementationMode = when {
-                        Build.VERSION.SDK_INT > 24 -> ImplementationMode.EXTERNAL
-                        else -> ImplementationMode.EMBEDDED
-                    },
-                    onRequestWindowColorMode = onRequestWindowColorMode,
-                    onTap = { x, y -> onTapToFocus(x, y) }
+                    implementationMode = implementationMode,
+                    coordinateTransformer = coordinateTransformer
                 )
             }
         }
     }
 }
 
+@Composable
+fun DetectWindowColorModeChanges(
+    surfaceRequest: SurfaceRequest,
+    implementationMode: ImplementationMode,
+    onRequestWindowColorMode: (Int) -> Unit
+) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val currentSurfaceRequest: SurfaceRequest by rememberUpdatedState(surfaceRequest)
+        val currentImplementationMode: ImplementationMode by rememberUpdatedState(
+            implementationMode
+        )
+        val currentOnRequestWindowColorMode: (Int) -> Unit by rememberUpdatedState(
+            onRequestWindowColorMode
+        )
+
+        LaunchedEffect(Unit) {
+            val colorModeSnapshotFlow =
+                snapshotFlow { Pair(currentSurfaceRequest.dynamicRange, currentImplementationMode) }
+                    .map { (dynamicRange, implMode) ->
+                        val isSourceHdr = dynamicRange.encoding != CXDynamicRange.ENCODING_SDR
```

Comment:
Is there any possible `isSourceHdr` would be not HDR Encoding, even it is not Encoding SDR, such as Encoding Dolby Vision, Encoding HLG, Encoding Unspecified?

Reply:
Dolby Vision, HLG10, HDR10 and HDR10+ are all considered HDR formats. Technically, the DynamicRange here from CameraX should never be UNSPECIFIED, since it is coming from `SurfaceRequest`. I'll make a note to update the CameraX documentation to better communicate that.

Reply:
The CameraX documentation improvements will be tracked with https://issuetracker.google.com/366536534

Reply:
Thank you for everything.